xrpl/models/auth_account.py

```diff
@@ -0,0 +1,70 @@
+"""Model used in AMMBid transaction."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Dict, Type
+
+from xrpl.models.base_model import BaseModel
+from xrpl.models.required import REQUIRED
+from xrpl.models.utils import require_kwargs_on_init
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class AuthAccount(BaseModel):
```

Comment:
This should inherit from `NestedModel` instead, which simplifies a lot of the logic